File: code_samples/dart/oop_abstraction.dart
// Oop: Abstraction
// AI/ML Training Sample

class Abstraction {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = Abstraction();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}

File: code_samples/go/oop_abstraction.go
package main

import (
    "fmt"
)

// Oop: Abstraction
// AI/ML Training Sample

type Abstraction struct {
    Data string
}

func NewAbstraction() *Abstraction {
    return &Abstraction{
        Data: "",
    }
}

func (s *Abstraction) Process(input string) {
    s.Data = input
}

func (s *Abstraction) Validate() bool {
    return len(s.Data) > 0
}

func (s *Abstraction) GetData() string {
    return s.Data
}

func main() {
    instance := NewAbstraction()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
